server/sonar-server/src/main/java/org/sonar/server/computation/component/ComponentImpl.java

```diff
@@ -83,6 +89,27 @@ public int getRef() {
     return component.getRef();
   }
 
+  public String getUuid() {
+    return uuid;
+  }
+
+  public ComponentImpl setUuid(String uuid) {
```

Comment:
I agree with you that we should use a immutable version ! But adding a @CheckForNull will force every caller to do some check and fail if null.
What about failing in the getUuid() if uuid is null ?